downstream_task_evaluation.py

```diff
@@ -626,13 +628,17 @@ def load_weights(
         pretrained_dict
     )  # v2 has the right para names
 
-    if is_dist:
```

Comment:
Might need to keep this bit in because sometimes we might want to load the weights that were just trained from one GPU :D

Reply:
But that's what `if 'module.' in list(pretrained_dict_v2.keys())[0]` already does, it essentially detects the value of is_dist. A single GPU trained module won't have the 'module' prefix in the layer names, a distributed trained model will. So simply checking if 'module' is the head tells you if it's distributed pretrained or not. There's no functional difference between single/multi GPU other than this prefix. That's why when you save a distributed model you should save the `model.module.state_dict()`, not `model.state_dict()`. This will automatically strip the 'module' prefix (and we wouldn't need is_dist in the first place).

So I think this makes is_dist obsolete? Or are there other scenarios where you may need to strip the head(s) of the keys?

Btw I pushed a new commit for this check. Converting the dict to list() is a bit wasterful, so changed it to an iterator and only get the first key. Also split the string to extract the actual head, to avoid errors if one of the layers happens to be named 'module'.